### PR TITLE
Adapted C++ blackboard client for the new blackboard architecture.

### PR DESCRIPTION
**Pull request by jwillebrands**

### CODE REVIEW

File: src/cpp/libraries/rexos_blackboard_cpp_client/src/BlackboardCppClient.cpp [hunk 1]
```diff
@@ -29,193 +30,146 @@
...
-				}
-			}
+
+int BlackboardCppClient::findDocuments(std::string queryAsJSON, std::vector<mongo::BSONObj> &results) {
+	mongo::ScopedDbConnection* connection = mongo::ScopedDbConnection::getScopedDbConnection(host);
+	std::string dbNamespace = database;
+	dbNamespace.append(".");
+	dbNamespace.append(collection);
+	mongo::Query query(mongo::fromjson(queryAsJSON));
+	std::auto_ptr<mongo::DBClientCursor> cursor = (*connection)->query(dbNamespace, query);
```
Comment (apvanzanten) [inline src/cpp/libraries/rexos_blackboard_cpp_client/src/BlackboardCppClient.cpp]: auto_ptr deprecated

File: src/cpp/libraries/rexos_blackboard_cpp_client/src/OplogMonitor.cpp [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+	std::cout << "Using query: " << query << std::endl;
+	return query;
+}
+
+void OplogMonitor::run()
+{
+	mongo::ScopedDbConnection* connection = mongo::ScopedDbConnection::getScopedDbConnection(host);
+
+	std::auto_ptr<mongo::DBClientCursor> tailedCursor = (*connection)->query(
```
Comment (apvanzanten) [inline src/cpp/libraries/rexos_blackboard_cpp_client/src/OplogMonitor.cpp]: auto_ptr deprecated

File: src/cpp/ros/equiplet_node/src/EquipletNode.cpp [hunk 3]
```diff
@@ -75,9 +84,10 @@
  *
  * @param json The message parsed in the json format
  **/
-void EquipletNode::blackboardReadCallback(std::string json){
+void EquipletNode::onMessage(Blackboard::BlackboardSubscription & subscription, const Blackboard::OplogEntry & oplogEntry) {
```
Comment (apvanzanten) [inline src/cpp/ros/equiplet_node/src/EquipletNode.cpp:85]: Comments need adjusting.

File: src/cpp/ros/equiplet_node/include/equiplet_node/EquipletNode.h [hunk 4]
```diff
@@ -114,5 +113,8 @@ class EquipletNode: BlackboardSubscriber{
 	 * @var BlackboardCppClient  *blackboardClient
 	 * Client to read from blackboard
 	 **/
-	BlackboardCppClient  *blackboardClient;
-};
\ No newline at end of file
+	Blackboard::BlackboardCppClient  *blackboardClient;
+
+	std::vector<Blackboard::BlackboardSubscription *> subscriptions;
```
Comment (apvanzanten) [inline src/cpp/ros/equiplet_node/include/equiplet_node/EquipletNode.h:121]: add comment

File: src/cpp/ros/equiplet_node/include/equiplet_node/EquipletNode.h [hunk 5]
```diff
@@ -38,24 +38,22 @@
...
+}
 /**
  * The equipletNode, will manage all modules and keep track of their states
  **/
-class EquipletNode: BlackboardSubscriber{
+class EquipletNode: Blackboard::BlackboardSubscriber{
+
 public:
 	EquipletNode(int id = 1);
 	virtual ~EquipletNode();
```
Comment (apvanzanten) [inline src/cpp/ros/equiplet_node/include/equiplet_node/EquipletNode.h:59]: void blackboardReadCallback() on line 60 should probably be removed.

File: src/cpp/ros/equiplet_node/src/EquipletNode.cpp (hunk 3 above)
Comment (jwillebrands) [inline src/cpp/ros/equiplet_node/src/EquipletNode.cpp:85]: Method is a stub and will be modified significantly during upcoming ROS development. I've removed incorrect parameters and made the description more generic. Considering the method does nothing at the moment I feel it's of no use to specify its workings in more detail.

**Comment by jwillebrands** [comment]
Despite successful testing, the client does not seem to be working properly. Closing pull-request until these problems have been resolved.

**Pull request closed without merging**